provider/data_source_wireguard_config_document.go

```diff
@@ -239,6 +254,7 @@ type WgConfig struct {
 	FirewallMark *string
 	WgQuickConfig
 	Peers []WgPeerConfig
+	Description *string
```

Comment:
I don't think `Description` is valid in a wg config file is it?

Reply:
It is acts as a descriptive marker in the terraform state file itself - easy to read at a glance when troubleshooting wireguard issue and need to verify/validate the key. 

It does not appear in the configuration file.

See this excerpt from the generated `terraform.tfstate` file and notice the `conf` attribute value, it never gets propagated into the value of the `conf`.

``` json
{
      "mode": "data",
      "type": "wireguard_config_document",
      "name": "server",
      "provider": "provider[\"registry.terraform.io/ojford/wireguard\"]",
      "instances": [
        {
          "schema_version": 0,
          "attributes": {
            "addresses": null,
            "conf": "\n[Interface]\nPrivateKey = iIquCC+BMaS73TYsS6CfEWYbjM9Hk2K3b3q8LL6wqX4=\nListenPort = 41240\nDNS = 10.100.0.1\nMTU = 1480\n\n\n# Peer # 0 is =\u003e Samsung S20\n[Peer]\nPublicKey = ngjJ/S98J2AXD4PHahGKn0UtJi7ZmQY5X4n3p0eHUyg=\nPresharedKey = 1nN52sdLAVyrNL7SMIimr5Mn6VgvQtI49LBC/zEVC1s=\nAllowedIPs = 0.0.0.0/0\nEndpoint = myserver.amazonaws.com:41240\nPersistentKeepalive = 25\n\n\n# Peer # 1 is =\u003e MBP 15\n[Peer]\nPublicKey = zeVoohx1oA76XovmvWtkl9zSwQjYSgnDVXegSM9tJgI=\nPresharedKey = RY3MDeBXsB3WrJycV6C+dhFKZ/prnCmKC067bgi9Z1Y=\nAllowedIPs = 0.0.0.0/0\nEndpoint = myserver.amazonaws.com:41240\nPersistentKeepalive = 25\n",
            "description": "My Wireguard Server",
            "dns": [
              "10.100.0.1"
            ],
            "firewall_mark": null,
            "id": "fcdAWZ/12/rGS6VRhvg6V9ISkjsEpfx/fzYUEcoOqxY=",
            "listen_port": 41240,
            "mtu": 1480,
            "peer": [
              {
                "allowed_ips": [
                  "0.0.0.0/0"
                ],
                "description": "Peer # 0 is =\u003e Samsung S20",
                "endpoint": "myserver.amazonaws.com:41240",
                "persistent_keepalive": 25,
                "preshared_key": "1nN52sdLAVyrNL7SMIimr5Mn6VgvQtI49LBC/zEVC1s=",
                "public_key": "ngjJ/S98J2AXD4PHahGKn0UtJi7ZmQY5X4n3p0eHUyg="
              },
              {
                "allowed_ips": [
                  "0.0.0.0/0"
                ],
                "description": "Peer # 1 is =\u003e MBP 15",
                "endpoint": "myserver.amazonaws.com:41240",
                "persistent_keepalive": 25,
                "preshared_key": "RY3MDeBXsB3WrJycV6C+dhFKZ/prnCmKC067bgi9Z1Y=",
                "public_key": "zeVoohx1oA76XovmvWtkl9zSwQjYSgnDVXegSM9tJgI="
              }
            ],
            "post_down": null,
            "post_up": null,
            "pre_down": null,
            "pre_up": null,
            "private_key": "iIquCC+BMaS73TYsS6CfEWYbjM9Hk2K3b3q8LL6wqX4=",
            "routing_table": null
          },
          "sensitive_attributes": [
            [
              {
                "type": "get_attr",
                "value": "private_key"
              }
            ]
          ]
        }
      ]
    }
```

Reply:
Why are the checks failing? Is there something that I can do to help?

Reply:
It's autoformatting from `go fmt` & `terraform fmt`, and docs generation from hashicorp/terraform-plugin-docs.

Don't worry too much about that though, I can always run on merge.

Sorry for the delay, I will get to this, I'm ok on the idea I think (though it's a little unconventional, since it's state-only, not something that affects the actual resource, not importable) just might need to tweak the examples a bit for aforementioned docs.